custom_model_runner/datarobot_drum/custom_task_interfaces/user_secrets.py

```diff
@@ -232,3 +253,125 @@ def get_dict(file_path: Path):
+    all_loggers = [logging.root.manager.root]
+    all_loggers.extend(logging.root.manager.loggerDict.values())
+    return all_loggers
+
+
+def _safely_add_filter_to_logger(logger, secrets_filter):
+    if hasattr(logger, "addFilter"):
+        logger.addFilter(secrets_filter)
+
+
+def _safely_remove_filters_from_logger(logger):
+    if not hasattr(logger, "filters") or not hasattr(logger, "removeFilter"):
+        return
+
+    for logger_filter in logger.filters:
+        if isinstance(logger_filter, SecretsScrubberFilter):
+            logger.removeFilter(logger_filter)
+
+
+class TextStreamSecretsScrubber:
+    def __init__(self, secrets: Iterable[AbstractSecret], stream: TextIO):
+        self.stream = stream
+        self.sensitive_data = get_ordered_sensitive_values(secrets)
+
+    def write(self, text):
+        new_text = scrub_values_from_string(self.sensitive_data, text)
+        self.stream.write(new_text)
+
+    def writelines(self, lines):
+        new_lines = [scrub_values_from_string(self.sensitive_data, text) for text in lines]
+        self.stream.writelines(new_lines)
+
+    def __getattr__(self, item):
+        return object.__getattribute__(self.stream, item)
+
+    def __eq__(self, other):
+        if not isinstance(other, TextStreamSecretsScrubber):
+            return False
+        return (self.stream, self.sensitive_data) == (other.stream, other.sensitive_data)
+
+
+class SecretsScrubberFilter(Filter):
+    def __init__(self, secrets: Iterable[AbstractSecret]):
+        super().__init__()
+        self.sensitive_data = get_ordered_sensitive_values(secrets)
+
+    def filter(self, record: LogRecord):
+        record.msg = self.transform(record.msg)
+        if isinstance(record.args, dict):
+            new_args = {k: self.transform(v) for k, v in record.args.items()}
+        elif isinstance(record.args, tuple):
+            new_args = tuple(self.transform(el) for el in record.args)
+        else:
+            new_args = self.transform(record.args)
+        record.args = new_args
+        return True
+
+    def transform(self, element):
+        if isinstance(element, str):
+            return scrub_values_from_string(self.sensitive_data, element)
+        else:
+            return element
+
+    def __eq__(self, other):
+        if not isinstance(other, SecretsScrubberFilter):
+            return False
+        return self.sensitive_data == other.sensitive_data
+
+
+def get_ordered_sensitive_values(secrets: Iterable[AbstractSecret]) -> List[str]:
+    """This returns the list of all sensitive values, including recursing through
+    sub-dictionaries so that they can be wiped from logs."""
+    if not secrets:
+        return []
+    values_generator = (_get_all_values(asdict(secret)) for secret in secrets)
+    empty_set: Set[str] = set()
+    all_values = empty_set.union(*values_generator)
+    longest_first_to_replace_both_strings_and_sub_strings = sorted(
+        all_values, key=lambda el: (-len(el), el)
```

Comment:
```suggestion
        all_values, key=lambda el: len(el), reverse=True
```

Reply:
ahhh!  missed it!  merged already :disappointed: if i remember after vacation ...